com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/app/discovery/ZigBeeNetworkDiscoverer.java

```diff
@@ -385,8 +389,14 @@ private boolean getAssociatedNodes(final int networkAddress) throws InterruptedE
             // Request extended response, start index for associated list is 0
             final IeeeAddressRequest ieeeAddressRequest = new IeeeAddressRequest(networkAddress, 1, startIndex);
             ieeeAddressRequest.setDestinationAddress(new ZigBeeEndpointAddress(networkAddress));
-            CommandResult response = networkManager.sendTransaction(ieeeAddressRequest, ieeeAddressRequest).get();
+            Future<CommandResult> resultFuture = networkManager.sendTransaction(ieeeAddressRequest, ieeeAddressRequest);
+            if (resultFuture == null) {
+                logger.debug("Transaction manager has been shutdown");
```

Comment:
I'm not 100% sure that this is correct? Firstly, I don't think that a null return here necessarily means that the transaction manager is shut down - although that is one possibility. Secondly though, if the transaction manager is shutdown and that is the cause of the null return, then this same message is already logged in `queueTransaction` so I'm not sure we need to have this log entry.

Reply:
Yeah, you are absolutely right. We can simply get rid of this log message.